core/src/main/java/org/infinispan/configuration/cache/AbstractStoreConfigurationBuilder.java

```diff
@@ -188,6 +189,15 @@ public void validate() {
       boolean preload = attributes.attribute(PRELOAD).get();
       boolean transactional = attributes.attribute(TRANSACTIONAL).get();
       ConfigurationBuilder builder = getBuilder();
+
+      Class storeClazz = attributes.getKlass();
+      if (storeClazz != null && storeClazz.isAnnotationPresent(Store.class)) {
```

Comment:
Thinking maybe if the annotation isn't present and shared is enabled that we log a WARN message saying that we can't verify if the cache store can be shared or not - as in it should be marked with @Store annotation.